### DISCUSSION TITLE
Unable to use ProrationModesAndroid.DEFERRED when switching between offers on the same subscription product

### DISCUSSION THREAD

**Discussion by jwoodmansey** [hyochan/react-native-iap#2474] [Q&A]
Hi,

I am pretty sure this is not an issue with react-native-iap itself so I'm not opening an issue but looking for some advice.

We are trying to enable users to switch between different offers on the same subscription on Android, however we are unable to use `ProrationModesAndroid.DEFERRED`. Although I can't find this anywhere in Google's docs I am assuming this isn't supported since an error (DEVELOPER_ERROR (5)) it thrown when calling `launchBillingFlow`. 

The issue with us using another method such as `IMMEDIATE_WITHOUT_PRORATION` is that we don't get informed that the user hasn't actually paid yet on the first server notification after switching to another offer on this subscription.

Essentially, the real-time developer notification wouldn't tell us that this first order wasn't really charged so our backend can't accurately track it.

<img width="870" alt="image" src="https://github.com/dooboolab-community/react-native-iap/assets/3459309/6ce847f9-b22f-4395-84f2-1f5b22395966">

How are others solving this situation?

**Comment by cervebar** [comment]
From our experience, I can confirm this for switching between base plans, `DEFERRED` for same base plans in one subscription in Billing library 5 is not supported by Google Android (and also not documented（＞д＜）).

When switching plans within the same subscription, only allowed proration modes are `IMMEDIATE_AND_CHARGE_FULL_PRICE` and `IMMEDIATE_WITHOUT_PRORATION`.  Empirically tested.

You have 2 options:
1) use one of this options, for us it was `IMMEDIATE_WITHOUT_PRORATION` as it seems more close to `DEFERRED` mode regarding the examples in [Android docs](https://developer.android.com/google/play/billing/subscriptions#proration) (the description of this mode itself, it is soooo confusing)

2) have 2 subscriptions for different time period and do not use base plans at all

I can expect it will be something similar for offers.

This discussion helped us:   https://github.com/android/play-billing-samples/issues/549#issuecomment-1179330667.
